libvast/src/system/index.cpp

```diff
@@ -684,46 +683,49 @@ index(index_actor::stateful_pointer<index_state> self,
         self->send(client, atom::done_v);
         return {};
       }
+      auto worker = self->state.next_worker();
+      if (!worker)
+        return caf::skip;
       // Get partition actors, spawning new ones if needed.
       auto actors
         = self->state.collect_query_actors(iter->second, num_partitions);
       // Send an evaluate atom to all the actors and collect the returned
       // evaluation triples in a `pending_query_map`, then run the continuation
       // below in the same actor context.
-      auto worker = self->state.next_worker();
       await_evaluation_maps(
         self, iter->second.expression, actors,
         [=](caf::expected<pending_query_map> maybe_pqm) {
           auto& st = self->state;
+          auto drop = [&] {
+            self->state.idle_workers.emplace_back(std::move(*worker));
```

Comment:
I think we should capture `worker = std::move(*worker)` in the lambda that we pass to `await_evaluation_maps`. This code does not need to know about the `optional` wrapping.